plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/AppEngineWebXmlValidatorTest.java

```diff
@@ -56,34 +83,43 @@ public void testValidate_badXml()
   public void testValidate_noBannedTags()
       throws IOException, CoreException, ParserConfigurationException {
     byte[] bytes = XML_NO_BANNED_ELEMENTS.getBytes(StandardCharsets.UTF_8);
-    ValidationResult result = AppEngineWebXmlValidator.validate(resource, bytes);
-    ValidatorMessage[] messages = result.getMessages();
-    assertEquals(0, messages.length);
+    AppEngineWebXmlValidator.validate(resource, bytes);
+    IMarker[] markers = resource.findMarkers(MARKER_TYPE, true, IResource.DEPTH_ZERO);
+    assertEquals(0, markers.length);
   }
 
   @Test
   public void testValidate()
       throws IOException, CoreException, ParserConfigurationException {
     byte[] bytes = XML.getBytes(StandardCharsets.UTF_8);
-    ValidationResult result = AppEngineWebXmlValidator.validate(resource, bytes);
-    ValidatorMessage[] messages = result.getMessages();
-    assertEquals(1, messages.length);
-    assertEquals(ELEMENT_MESSAGE, (String) messages[0].getAttribute(IMarker.MESSAGE));
+    AppEngineWebXmlValidator.validate(resource, bytes);
+    IMarker[] markers = resource.findMarkers(MARKER_TYPE, true, IResource.DEPTH_ZERO);
+    assertEquals(1, markers.length);
+    assertEquals(ELEMENT_MESSAGE, (String) markers[0].getAttribute(IMarker.MESSAGE));
   }
   
   @Test
-  public void testCreateMessage() {
+  public void testCreateMarker() throws CoreException {
     BannedElement element = new BannedElement(ELEMENT_NAME);
-    ValidatorMessage message = AppEngineWebXmlValidator.createMessage(resource, element, 0);
-    assertEquals(ELEMENT_MESSAGE, (String) message.getAttribute(IMarker.MESSAGE));
+    AppEngineWebXmlValidator.createMarker(resource, element, 0);
+    IMarker[] markers = resource.findMarkers(MARKER_TYPE, true, IResource.DEPTH_ZERO);
+    assertEquals(ELEMENT_MESSAGE, (String) markers[0].getAttribute(IMarker.MESSAGE));
   }
   
   @Test
-  public void testCreateSAXErrorMessage() {
+  public void testCreateSAXErrorMessage() throws CoreException {
     SAXParseException spe = new SAXParseException("", "", "", 1, 1);
     SAXException ex = new SAXException(ELEMENT_MESSAGE, spe);
-    ValidationResult result = AppEngineWebXmlValidator.createSaxErrorMessage(resource, ex);
-    ValidatorMessage[] messages = result.getMessages();
-    assertEquals(ELEMENT_MESSAGE, (String) messages[0].getAttribute(IMarker.MESSAGE));
+    AppEngineWebXmlValidator.createSaxErrorMessage(resource, ex);
+    IMarker[] markers = resource.findMarkers(MARKER_TYPE, true, IResource.DEPTH_ZERO);
+    assertEquals(ELEMENT_MESSAGE, (String) markers[0].getAttribute(IMarker.MESSAGE));
+  }
+  
+  private static void createFolders(IContainer parent, IPath path) throws CoreException {
```

Comment:
This is duplicate code of `StandardFacetInstallation.createFolders()`. Since this is generic, we could move this to the util bundle.